modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_3_2/TemplateBrowserSnifferUpgradeProcess.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v5_3_2;
+
+import com.liferay.dynamic.data.mapping.internal.upgrade.BaseTemplateUpgradeProcess;
+import com.liferay.petra.string.StringPool;
+
+import java.util.regex.Pattern;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+public class TemplateBrowserSnifferUpgradeProcess
```

Comment:
We do have, @marcelabc!

Basically, since the most recent portal versions, a few variables started to be added by default in the context of the FreeMarker templates, please see all these variables in the method [prepare](https://github.com/liferay/liferay-portal/blob/f63af3907491f652f2c1eed4216b63825f26b1fc/modules/apps/portal-template/portal-template-engine-api/src/main/java/com/liferay/portal/template/engine/TemplateContextHelper.java#L202) of the class TemplateContextHelper. This pull aims to create a foundation to make easier the adding of the new cases like the variable browser sniffer.

Does it make sense to you?

Reply:
Thank you for the explanation! Yes, it makes sense :+1: